src/simularium/RemoteSimulator.ts

```diff
@@ -255,56 +266,61 @@ export class RemoteSimulator implements ISimulator {
+    ): Promise<boolean> {
+        // Check if the WebSocket becomes connected within an allotted amount
+        // of time and number of retries.
+
+        // Initially wait for a max wait time of maxWaitTime, then retry
+        // connecting <maxRetries> time(s). In a retry, only wait for the
+        // amount of time specified as timeout.
+
+        const maxWaitTime = 4 * timeout;
+
+        const isConnected = await this.waitForWebSocket(timeout);
+        this.connectionTimeWaited += timeout;
+
+        if (isConnected) {
+            return true;
+        } else if (this.connectionTimeWaited < maxWaitTime) {
+            return this.checkConnection(address, timeout);
+        } else if (this.connectionRetries < maxRetries) {
+            this.createWebSocket(address);
+            this.connectionRetries++;
+            return this.checkConnection(address, timeout);
+        } else {
+            return false;
+        }
     }
 
-    public connectToRemoteServer(address: string): Promise<string> {
-        const remoteStartPromise = new Promise<string>((resolve, reject) => {
-            if (this.socketIsConnected()) {
-                return resolve("Remote sim successfully started");
-            }
+    public async connectToRemoteServer(address: string): Promise<string> {
+        this.connectionTimeWaited = 0;
+        this.connectionRetries = 0;
 
-            const startPromise = this.connectToUriAsync(address);
-            // wait 1 second for websocket to open
-            const waitForIsConnected = () =>
-                new Promise((resolve) =>
-                    setTimeout(() => {
-                        resolve(this.socketIsConnected());
-                    }, 1000)
-                );
+        if (this.socketIsConnected()) {
+            return CONNECTION_SUCCESS_MSG;
+        }
 
-            const handleReturn = async () => {
-                const isConnected = await waitForIsConnected();
-                secondsWaited++;
-                if (isConnected) {
-                    resolve("Remote sim successfully started");
-                } else if (secondsWaited < TOTAL_WAIT_SECONDS) {
-                    return await handleReturn();
-                } else if (connectionTries <= MAX_CONNECTION_TRIES) {
-                    connectionTries++;
-                    return this.connectToUriAsync(address).then(handleReturn);
-                } else {
-                    reject(
-                        new Error(
-                            "Failed to connected to requested server, try reloading. If problem keeps occurring check your connection speed"
-                        )
-                    );
-                }
-            };
-            const TOTAL_WAIT_SECONDS = 2;
-            let secondsWaited = 0;
-            const MAX_CONNECTION_TRIES = 2;
-            let connectionTries = 1;
-            return startPromise.then(handleReturn);
-        });
+        this.createWebSocket(address);
```

Comment:
Is it possible to move this line inside of this.checkConnection? (is checkConnection called by anything else?)